### PR TITLE
Supporting full-fledged URLs for authelia URL (instead of domain:port)

### PR DESCRIPTION
**Pull request by clems71**
- Fixes the case where authelia is hosted on a path, not a subdomain (like cloud.example.com/auth)
- Simplifies the handler: no need for TLS flag, neither port : relying on URL now.
- Improved doc

### CODE REVIEW

**Comment by HeavenVolkoff** [comment]
Hey, thanks for your interest in this project and for the PR.
I gave your changes a quick look yesterday, and they look good to me.
Today or tomorrow I will test them out, and if everything is working fine, I will merge this

File: README.md [hunk 1]
```diff
@@ -10,31 +10,40 @@ Caddy 2 plugin for integration with Authelia
...
+        request_header Remote-User {http.auth.user.id}
+        request_header Remote-Groups {http.auth.user.groups}
         reverse_proxy http://whoami
     }
 }
 
 authelia.example.com {
-	reverse_proxy http://authelia:9091
+    # Needed for 2FA to work - authelia checks X-Forwarded-Host, and Caddy doesn't fill it by default
+    request_header X-Forwarded-Host {http.request.host}
```
Comment (HeavenVolkoff) [inline README.md:33]: This shouldn't be necessary. The code below should already be assigning all the headers needed by the authelia `/api/verify` request:
https://github.com/HeavenVolkoff/caddy-authelia/blob/ca9e35288d873885443b3dc3669d4d3de0dc0ea3/plugin/internalized/traefik/write_headers.go#L50-L58

Is this not working for you?

**Comment by clems71** [comment]
Hey! Did you had a chance to test the PR? I'm about to rebuild a caddy binary with your plugin, so just wondering!

**Comment by HeavenVolkoff** [comment]
> Hey! Did you had a chance to test the PR? I'm about to rebuild a caddy binary with your plugin, so just wondering!

Hey, sorry I wasn't able to test it yet, had some family issues. But start of next week I will have some time to test it properly.

By the way I left a comment about the `X-Forwarded-Host` header rule in your README, did you read it?

**Comment by clems71** [comment]
Hey, no problem. No didn't read the comment. I double checked, and couldn't find it on Github. Maybe still in draft on your end ?
Regards

**Review by HeavenVolkoff — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (HeavenVolkoff) [reply README.md:33]: Ok, now I understand the problem here. The AssignForwardHeaders function is only executed during the api calls made to authelia. But these headers seems to also be required when you open the authelia frontend to sign-in with 2FA, and accessing that goes through another set of caddy directives which this plugin does not handle. I will look for a way to solve this automatically with the plugin, but for now this seems to be the best solution.

**Comment by HeavenVolkoff** [comment]
LGTM

I have done some minimal tests, and everything seems fine.

In the future I will try to write some github action to test this plugin automatically.

Anyway, thanks for you contribution.

**Review by HeavenVolkoff — APPROVED** [review]

**Pull request merged**